teuthology/orchestra/remote.py

```diff
@@ -229,20 +229,28 @@ def mkdtemp(self, suffix=None, parentdir=None):
 
         return self.sh(args).strip()
 
-    def mktemp(self, suffix=None, parentdir=None):
+    def mktemp(self, suffix=None, parentdir=None, path=None, data=None,
```

Comment:
I don't understand why do you need path argument? The mktemp should care about the path, if user wants to use path, the write_file should be used?
Also, could you please explain why owner is required?